spectator/meter/counter.go

```diff
@@ -34,16 +34,24 @@ func (c *Counter) Increment() {
 	c.writer.Write(line)
 }
 
-// AddFloat adds a specific float64 delta to the current measurement.
+// Add adds an int64 delta to the current measurement.
+func (c *Counter) Add(delta int64) {
+	if delta > 0 {
+		var line = fmt.Sprintf("%s:%s:%d", c.meterTypeSymbol, c.id.spectatordId, delta)
+		c.writer.Write(line)
+	}
+}
+
+// AddFloat adds a float64 delta to the current measurement.
 func (c *Counter) AddFloat(delta float64) {
 	if delta > 0.0 {
 		var line = fmt.Sprintf("%s:%s:%f", c.meterTypeSymbol, c.id.spectatordId, delta)
 		c.writer.Write(line)
 	}
 }
 
-// Add is to add a specific int64 delta to the current measurement.
-func (c *Counter) Add(delta int64) {
+// AddUint adds an uint64 delta to the current measurement.
+func (c *Counter) AddUint(delta uint64) {
 	if delta > 0 {
```

Comment:
Can a UInt ever be < 0?

Reply:
This check was overlooked. That being said, this PR has been reworked to focus on `MonotonicCounter` alone.